### PR TITLE
Added arrays exercises

### PR DESCRIPTION
**Pull request by mathias-onu**
Finishing the "Arrays" chapter from the book "Data Structures and Algorithms with JavaScript", I'm pushing the solutions for the exercises at the end of the chapter.

### CODE REVIEW

File: src/ds-algorithms/arrays.ts [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        this.letters = word.split('')
+    }
+
+    singleWord(): string {
+        return this.letters.join('')
+    }
+}
+
+const animal = new Letters('dog')
+animal.letters
```
Comment (pancudaniel7) [inline src/ds-algorithms/arrays.ts]: - Question: Are rost linia asta de cod aici?

Comment (mathias-onu) [reply src/ds-algorithms/arrays.ts]: Nu, nu prea are rost - era pusă într-un console.log() - să o șterg?

Comment (pancudaniel7) [reply src/ds-algorithms/arrays.ts]: Tu ce zici, de exemplu Golang nu iti compileaza daca ai variablile care nu le folosesti...

Comment (mathias-onu) [reply src/ds-algorithms/arrays.ts]: Da, e useless linia - o voi șterge.

File: src/ds-algorithms/arrays.ts [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    average(): number {
+        let total: number = 0
+
+        for(let i = 0; i < this.grades.length; i++) {
+            total += this.grades[i]
+        }
+
+        return total / this.grades.length
+    }
+}
```
Comment (pancudaniel7) [inline src/ds-algorithms/arrays.ts:32]: **Advice**:
- It will be interesting to use encapsulation here

Comment (pancudaniel7) [reply src/ds-algorithms/arrays.ts:32]: Here you have an example: https://1.bp.blogspot.com/-dDN_hHIh6H8/XRSoOYMJtpI/AAAAAAAAGJI/5y9-J8eEe4AIeGrRIEJBTlZIpUrKmbMeQCLcBGAs/s1600/generate-getter-setter-typescript.gif

**Comment by pancudaniel7** [comment]
If you tested it and it works I don't have anything to add besides the current comments.

File: src/ds-algorithms/arrays.ts (hunk 2 above)
Comment (mathias-onu) [reply src/ds-algorithms/arrays.ts:32]: 👍 I've refactored my code to adhere to the encapsulation concept. Can you please take another look? 🙏

Comment (pancudaniel7) [reply src/ds-algorithms/arrays.ts:32]: Every time you finish your comment fixes you can hit resolve button

Comment (mathias-onu) [reply src/ds-algorithms/arrays.ts:32]: Ok, noted 👍

**Review by pancudaniel7 — APPROVED** [review]
Looks good!

**Pull request merged**